src/webgpu/api/validation/capability_checks/features/texture_formats.spec.ts

```diff
@@ -171,6 +171,53 @@ g.test('canvas_configuration')
     }
   });
 
+g.test('canvas_configuration_view_formats')
+  .desc(
+    `
+  Test that configuring a canvas with view formats throws an exception if the required optional
+  feature is not enabled. Otherwise, a validation error should be generated instead of throwing an
+  exception.
+  `
+  )
+  .params(u =>
+    u
+      .combine('format', kOptionalTextureFormats)
+      .combine('canvasType', kAllCanvasTypes)
+      .combine('enable_required_feature', [true, false])
+  )
+  .beforeAllSubcases(t => {
+    const { format, enable_required_feature } = t.params;
+
+    const formatInfo = kTextureFormatInfo[format];
+    if (enable_required_feature) {
+      t.selectDeviceOrSkipTestCase(formatInfo.feature);
```

Comment:
Here, you can use `selectDeviceForTextureFormatOrSkipTestCase` which can take an array of formats, so if you do the `viewFormats` thing above then you can pass it in here directly.

Reply:
ok, I update this test to use it for an array of formats.